JavaScript/JavaScriptSDK/Sender.ts

```diff
@@ -407,6 +419,29 @@ module Microsoft.ApplicationInsights {
         }
 
         /**
+         * Send Beacon API request
+         * @param payload {string} - The data payload to be sent.
+         * @param isAsync {boolean} - not used
+         */
+        private _beaconSender(payload: string[], isAsync: boolean) {
+            var url = this._config.endpointUrl();
+            var batch = this._buffer.batchPayloads(payload);
+
+            // The sendBeacon method returns true if the user agent is able to successfully queue the data for transfer. Otherwise it returns false.
+            var queued = (<any>navigator).sendBeacon(url, batch);
+
+            if (queued) {
+                this._buffer.markAsSent(payload);
+            } else {
+                _InternalLogging.throwInternalNonUserActionable(LoggingSeverity.WARNING,
+                    new _InternalLogMessage(_InternalMessageId.NONUSRACT_TransmissionFailed, ". " +
+                        "Failed to send telemetry with Beacon API."));
+
+                // consider switching back to XHR
```

Comment:
My point was that we need to give visibility for a customer that after the switch to beacon so much telemetry got lost... Or something along this lines. Maybe even make it "error" instead of "warning"